Extensions/TileMap/simpletilemapruntimeobject.ts

```diff
@@ -0,0 +1,764 @@
+    _renderer: gdjs.TileMapRuntimeObjectPixiRenderer;
+    _rowCount: number;
+    _columnCount: number;
+    _tileSize: number;
+    _displayMode = 'all';
+    _layerIndex = 0;
+    _initialTileMapAsJsObject: object | null = null;
+    _initialTilesWithHitBox: number[];
+    _isTileMapDirty: boolean = false;
+    _sceneToTileMapTransformation: gdjs.AffineTransformation = new gdjs.AffineTransformation();
+    _tileMapToSceneTransformation: gdjs.AffineTransformation = new gdjs.AffineTransformation();
+    _collisionTileMap: gdjs.TileMap.TransformedCollisionTileMap | null = null;
+    _hitBoxTag: string = 'collision';
+    private _transformationIsUpToDate: boolean = false;
+
+    // TODO: Add a debug mode like for TileMapCollisionMaskRuntimeObject to draw?
+
+    constructor(
+      instanceContainer: gdjs.RuntimeInstanceContainer,
+      objectData: SimpleTileMapObjectDataType
+    ) {
+      super(instanceContainer, objectData);
+      this._opacity = objectData.content.opacity;
+      this._atlasImage = objectData.content.atlasImage;
+      this._rowCount = objectData.content.rowCount;
+      this._columnCount = objectData.content.columnCount;
+      this._tileSize = objectData.content.tileSize;
+      this._initialTilesWithHitBox = (objectData.content
+        .tilesWithHitBox as string)
+        .split(',')
+        .filter((id) => !!id)
+        .map((idAsString) => parseInt(idAsString, 10));
+      this._tileMapManager = gdjs.TileMap.TileMapRuntimeManager.getManager(
+        instanceContainer
+      );
+      this._renderer = new gdjs.TileMapRuntimeObjectRenderer(
+        this,
+        instanceContainer
+      );
+
+      // *ALWAYS* call `this.onCreated()` at the very end of your object constructor.
+      this.onCreated();
+    }
+
+    getRendererObject() {
+      return this._renderer.getRendererObject();
+    }
+
+    updatePreRender(instanceContainer: gdjs.RuntimeInstanceContainer): void {
+      if (this._isTileMapDirty) {
+        this._tileMapManager.getOrLoadSimpleTileMapTextureCache(
+          (textureName) => {
+            return (this.getInstanceContainer()
+              .getGame()
+              .getImageManager()
+              .getPIXITexture(textureName) as unknown) as PIXI.BaseTexture<
+              PIXI.Resource
+            >;
+          },
+          this._atlasImage,
+          this._tileSize,
+          this._columnCount,
+          this._rowCount,
+          (textureCache: TileMapHelper.TileTextureCache | null) => {
+            if (!textureCache) {
+              // getOrLoadTextureCache already log warns and errors.
+              return;
+            }
+            this._renderer.refreshPixiTileMap(textureCache);
+          }
+        );
+        if (this._collisionTileMap) {
+          const tileMap = this._renderer.getTileMap();
+          if (tileMap) this._collisionTileMap.updateFromTileMap(tileMap);
+        }
+        this._isTileMapDirty = false;
+      }
+    }
+
+    updateFromObjectData(
```

Comment:
I wonder if a comment like "Map content is updated at hot-reload by extraInitializationFromInitialInstance" could be useful.